doc/nrf/releases_and_maturity/releases/release-notes-2.5.1.rst

```diff
@@ -0,0 +1,126 @@
+See :ref:`cloning_the_repositories` and :ref:`gs_updating_repos_examples` for more information.
+
+For information on the included repositories and revisions, see `Repositories and revisions for v2.5.1`_.
+
+IDE and tool support
+********************
+
+`nRF Connect for Visual Studio Code extension <nRF Connect for Visual Studio Code_>`_ is the only officially supported IDE for |NCS| v2.5.1.
+
+:ref:`gs_app_tcm`, used to install the |NCS| automatically from `nRF Connect for Desktop`_, is available for Windows, Linux, and macOS.
+
+Supported modem firmware
+************************
+
+See `Modem firmware compatibility matrix`_ for an overview of which modem firmware versions have been tested with this version of the |NCS|.
+
+Use the latest version of the nRF Programmer app of `nRF Connect for Desktop`_ to update the modem firmware.
+See :ref:`nrf9160_gs_updating_fw_modem` for instructions.
+
+Modem-related libraries and versions
+====================================
+
+.. list-table:: Modem-related libraries and versions
+   :widths: 15 10
+   :header-rows: 1
+
+   * - Library name
+     - Version information
+   * - Modem library
+     - `Changelog <Modem library changelog for v2.5.1_>`_
+   * - LwM2M carrier library
+     - `Changelog <LwM2M carrier library changelog for v2.5.1_>`_
+
+Known issues
+************
+
+Known issues are only tracked for the latest official release.
+See `known issues for nRF Connect SDK v2.5.1`_ for the list of issues valid for the latest release.
+
+.. _ncs_release_notes_251_changelog:
+
+Changelog
+*********
+
+The following sections provide detailed lists of changes by component.
+
+Protocols
+=========
+
+This section provides detailed lists of changes by :ref:`protocol <protocols>`.
+
+Wi-Fi
+-----
+
+* Added improvements and fixes related to the following:
+
+  * :term:`Delivery Traffic Indication Message (DTIM)` power save functionality
+  * Firmware issues due to AP power cycle
+  * MAC address configuration
+  * General system stability
+
+nrfx
+====
+
+* Updated the nrfx QSPI driver behavior.
+  Now, QSPI peripheral is not activated during the driver initialization but rather when the first operation is requested or when the :c:func:`nrfx_qspi_activate` function is called.
+* Fixed an issue when external memory requires a relatively long time for the operation to happen.
+  This is done by adding a timeout member to :c:struct:`nrfx_qspi_config_t` , which can be configured at runtime.
+  If the timeout value is ``0``, the old value is used instead (500 milliseconds).
+* Removed the ongoing custom instruction long transfer check from the deactivate function.
+  The check could trigger errors, and now you must make sure that the custom instruction long transfer has ended, otherwise it will be interrupted.
+  This change affects the behavior of the :c:func:`nrfx_qspi_uinit` and the :c:func:`nrfx_qspi_deactivate` functions.
```

Comment:
```suggestion
  This change affects the behavior of the :c:func:`nrfx_qspi_uinit` and :c:func:`nrfx_qspi_deactivate` functions.
```
Might be good to have a bit further information on how the change affects the behavior of the functions.

Reply:
Maybe it's unclear but it's described above. The check for ongoing custom instruction has been removed so now when the user calls `nrfx_qspi_uninit` or `nrfx_qspi_deactivate` there is a chance that this will interrupt ongoing custom instruction transfer as it's not checked anymore

Reply:
Right, at least it was not clear to me. Maybe the nrfx_qspi_deactivate function should be mentioned in the first sentence of the bullet item. The last sentence with the changed behavior of these two functions feels a bit off.

Reply:
I have rephrased the sentence a bit as per the above comment. Please check.